kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/NodeShapeImpl.java

```diff
@@ -66,5 +67,13 @@ public void applyPosition(final Node<View<W>, Edge> element,
     @Override
     public void applyState(final ShapeState shapeState) {
         getShape().applyState(shapeState);
+        ShapeView shape = getShapeView();
+        if (!isSelected() && shape instanceof HasControlPoints) {
+            ((HasControlPoints<Object>)shape).hideControlPoints();
+        }
+    }
+
+    private boolean isSelected() {
+        return ShapeState.SELECTED.equals(getShape().getShapeStateHandler().getShapeState());
```

Comment:
I understand it is not a case currently, but to avoid any chance of NPE in future, I usually ask in reviews to use:
```
Objects.equals(ShapeState.SELECTED, getShape().getShapeStateHandler().getShapeState());
```

Reply:
Yea, I think it's low probability of null pointer there. But I see your point, updated, thank you!